tests/lib/state.sh

```diff
@@ -0,0 +1,123 @@
+#!/bin/bash
+
+SNAPD_STATE_PATH="$SPREAD_PATH"/snapd-state
+SNAPD_STATE_FILE="$SPREAD_PATH"/snapd-state.tar
+
+# shellcheck source=tests/lib/dirs.sh
+. "$TESTSLIB/dirs.sh"
+
+# shellcheck source=tests/lib/boot.sh
+. "$TESTSLIB/boot.sh"
+
+delete_snapd_state() {
+    rm -rf $SNAPD_STATE_PATH
+    rm -f $SNAPD_STATE_FILE
+}
+
+save_classic_state() {
+    escaped_snap_mount_dir=$1
+    snapd_env="/etc/environment /etc/systemd/system/snapd.service.d /etc/systemd/system/snapd.socket.d"
+    snap_confine_profiles="$(ls /etc/apparmor.d/snap.core.* || true)"
+    # shellcheck disable=SC2086
+    tar cf "$SNAPD_STATE_FILE" /var/lib/snapd "$SNAP_MOUNT_DIR" /etc/systemd/system/"$escaped_snap_mount_dir"-*core*.mount /etc/systemd/system/multi-user.target.wants/"$escaped_snap_mount_dir"-*core*.mount $snap_confine_profiles $snapd_env
+}
+
+restore_classic_state() {
```

Comment:
I think it would be simpler if we had just one function that does the right thing per system.